### ISSUE TITLE
Upload photos

### ISSUE THREAD

**Issue by yesahem** [yesahem/Medium-Blog#9]
Add a feature where user can upload photos and readers can see the photos along with the posts

**Comment by saura8668** [comment]
Hi @yesahem, can you please assign me this issue?

**Comment by yesahem** [comment]
Go ahead @saura8668

**Comment by geekyharsh05** [comment]
@saura8668 Any updates?

**Comment by saura8668** [comment]
Hi @geekyharsh05, I have done the frontend. But facing issues while storing it into database. I've created the wranler.toml file to store the JWT_Token and Database_url but it is not fetching from that file. Can you help me with the setup?

**Comment by geekyharsh05** [comment]
@saura8668 yeah sure. Can you please tell me have you forked and cloned the newer dev branch or the older one?

**Comment by saura8668** [comment]
Yeah I'm using the older one. It's showing my fork is 33commit behind. 
This branch is [33 commits behind](https://github.com/saura8668/Medium-Blog/compare/main...yesahem%3AMedium-Blog%3Amain) yesahem/Medium-Blog:main.

this is showing

**Comment by geekyharsh05** [comment]
@saura8668 Delete your fork and fork it again then clone it.

**Comment by saura8668** [comment]
done

**Comment by geekyharsh05** [comment]
Now go to neon.tech grab a database url and afterwards go to prisma accelerate and generate an accelerate url.

**Comment by saura8668** [comment]
Got a DataBaseURL from prisma

**Comment by geekyharsh05** [comment]
> Got a DataBaseURL from prisma
> 

Now paste it in the env file of your backend folder

**Comment by saura8668** [comment]
Done

**Comment by geekyharsh05** [comment]
@saura8668 after pasting it in the env file then run the following commands:
```bash
npm run postinstall
```

```bash
npm run prisma:migrate
```

```bash
npm run dev
```

**Comment by saura8668** [comment]
Done

**Comment by geekyharsh05** [comment]
@saura8668 Is it working now?

**Comment by saura8668** [comment]
Object { name: "PrismaClientInitializationError", clientVersion: "5.20.0" }
​
clientVersion: "5.20.0"
​
name: "PrismaClientInitializationError"

getting this error after I try to signup using the UI

getting the DATABASE_URL undefined after it hits the signup endpoint
Further I have pasted the DATABASE_URL in [.env.example] this file and renamed as .env

**Comment by geekyharsh05** [comment]
@saura8668 Is your .env file is in the backend folder? Or it is in the root.

**Comment by geekyharsh05** [comment]
@saura8668 Make sure it is inside the backend folder.

**Comment by saura8668** [comment]
@geekyharsh05 yes I've created it inside backend folder.still getting undefined for c.env.DATABASE_URL

**Comment by geekyharsh05** [comment]
@saura8668 Connect with me on [X](https://x.com/geekyharsh05).

**Comment by yesahem** [comment]
@saura8668 schedule a meet with me or @geekyharsh05 day after tomorrow would love to help !!!
discord(mine): ahemraj
x(mine): [X](https://www.x.com/heyshishu)

**Comment by saura8668** [comment]
Sure. @geekyharsh05 I've pinged you in X

**Comment by saura8668** [comment]
Hi @yesahem, @geekyharsh05 I was able to solve the undefined environment variable issue. The frontend is ready just need to do the backend part to store images.

**Comment by yesahem** [comment]
> Hi @yesahem, @geekyharsh05 I was able to solve the undefined environment variable issue. The frontend is ready just need to do the backend part to store images.

Do you need any more help regarding the backend or it is fine now !!
Let me know 


If you need any help ping me over[ X](https://x.com/heyshishu)

**Comment by yesahem** [comment]
Any updates @saura8668 ??

**Comment by saura8668** [comment]
Hi @yesahem, I am able to store pictures in my local using cloudflare r2 workers. But having some problems in deployments. Lets have a call tomorrow.
![image](https://github.com/user-attachments/assets/ce5ca1b4-10d7-4ec0-9a97-fba4ed63ada3)

**Comment by yesahem** [comment]
Hey @saura8668 hit me over discord : 
Discord: ahemraj

**Comment by saura8668** [comment]
Hi @yesahem / @geekyharsh05, please review the below PR:
https://github.com/yesahem/Medium-Blog/pull/58

Let me know if any changes required

[Screencast from 24-10-24 11:46:03 PM IST.webm](https://github.com/user-attachments/assets/a220e2b4-ba79-4d8b-add8-cf14cacc2fc9)

**Comment by yesahem** [comment]
@saura8668 
Why is is coming under comment section??

And can you please store the image over to cloudinary(or any cloud provider) and then store the cloudiniary_image_id in the db and when user open a post hits the db get-back the image_id and then hit the cloudinary url with image id to show photo to the user!


Storing image in the folder itself is issue I am saying to fix that's all

**Comment by saura8668** [comment]
Hi @yesahem , I've used cloudflare R2 bucket (online) to store all the images and its id is stored in postgresql db which was created in Neon.tech site.

**Comment by yesahem** [comment]
Hii @saura8668 please also answer the first question

Why is is coming under comment section??

> @saura8668 Why is is coming under comment section??
> 
> And can you please store the image over to cloudinary(or any cloud provider) and then store the cloudiniary_image_id in the db and when user open a post hits the db get-back the image_id and then hit the cloudinary url with image id to show photo to the user!
> 
> Storing image in the folder itself is issue I am saying to fix that's all

**Comment by saura8668** [comment]
Hi @yesahem, the user should upload the pictures while posting.....that what I understood hence I added the pictures section after the Post section. Please correct me, If I misunderstood something.

**Comment by yesahem** [comment]
yeah you are right!!!

> Hi @yesahem, the user should upload the pictures while posting.....that what I understood hence I added the pictures section after the Post section. Please correct me, If I misunderstood something.

**Comment by saura8668** [comment]
Hi, @yesahem please review the pr and if everything is working properly then can you please merge it. Let me know if you need any more changes.

**Comment by yesahem** [comment]
> Hi, @yesahem please review the pr and if everything is working properly then can you please merge it. Let me know if you need any more changes.

Hey @saura8668 please fix the conficts

**Comment by saura8668** [comment]
@yesahem, please check now
